tests/bugfixes/github/test_issue_987.py

```diff
@@ -0,0 +1,28 @@
+# -*- coding: utf-8 -*-
+
+from system_tests import CaseMeta, path
+import zlib
+
+class tagListTest(metaclass=CaseMeta):
+    def compare_stdout(self, i, command, got_stdout, expected_stdout):
+        lines    = got_stdout.split('\n')
+        count    = len(lines);
+        checksum = zlib.adler32(got_stdout.encode())
+        new_got  = "%s,%s" % (count, checksum)
+        return super().compare_stdout(i, command, new_got, expected_stdout)
+
+    commands = ["$taglist --help",
+                "$taglist all",
+                "$taglist ALL",
+                "$taglist Groups",
+                "$taglist nothing",
+               ]
+    stdout =   ["8,198867862",
```

Comment:
Let's not do this in master, this completely obscures what is actually going on.

Reply:
You can do as you wish in v0.28.

I used the "adler32" approach to avoid adding a large (>1 mb) stdout reference output.  I originally proposed the argument --adler32 in the `taglist` command and you requested that functionality in the test suite.

Reply:
What's your proposal then @D4N ?

For me, it looks a bit over killing to be testing the `taglist` application, which is part of the **samples**.Furthermore, this test can be considered as "brittle" since it will break everytime we add support for new tags ...

Reply:
The test in itself is not too bad: `taglist` just outputs all tags known to exiv2. Imho it would be quite useful to have a check that a tag got added or modified.

The issue is though: there's a lot of them and so it was opted to check their checksum instead (which changes every time you touch a tag, but obscures the actual change).

If we can keep the output of the taglist application at a manageable size, then I'd say: check the output directly. Otherwise let's kill this test.

Reply:
Yes, I remember when that decision was taken and it did not seem a bad idea at that moment. Now, what we are observing is that the error triggered by the test are obscure, and only someone with the proper background will understand what's happening. 

I think we could try to recover the real output of the taglist application in plain test so that it is easy to understand why the test fails. If we see that it is too much text or it is difficult to anyways figure out what it is failing, we could drop the test as a 2nd option.

If you agree with this approach, please give me a :+1: and I'll create another PR to deal with the situation.

Reply:
Having the tags in plain text would be imho the best solution!